scripts/executors/src/start.ts

```diff
@@ -0,0 +1,294 @@
+// eslint-disable-next-line @typescript-eslint/triple-slash-reference -- cannot import .d.ts as that is causing failures within ts-node/register
+/// <reference types="./enquirer-types.d.ts" />
+
+import { spawnSync } from 'node:child_process';
+
+import { createProjectGraphAsync, output } from '@nx/devkit';
+import type { ProjectGraph, ProjectGraphProjectNode } from '@nx/devkit';
+import type { Choice } from 'enquirer';
+import { AutoComplete } from 'enquirer';
+
+const projectTitleDefault = 'Fluent UI';
+const repoNpmNameDefault = 'fluentui-repo';
+const projectHeaderDefault = {
+  projectNameCell: 'Project Name',
+  descriptionCell: '   Type   /      Scope      /      Access      ',
+};
+const defaultTargetDescription = {
+  build: 'transpile source TS to JS',
+  'bundle-size': 'check the bundle size diff',
+  bundle: 'bundle source with webpack',
+  storybook: 'start storybook in dev mode',
+  'build-storybook': 'build production version of storybook',
+  'generate-api': 're-generate api.md (using api-extractor)',
+  lint: 'validate code with eslint',
+  'type-check': 'validate type correctness via TypeScript',
+  test: 'run jest tests ',
+  'test-integration': 'run integration tests (using @fluentui/scripts-projects-test)',
+  'test-ssr': 'run server-side rendering tests (using @fluentui/scripts-test-ssr)',
+  'test-vr': 'run visual regression tests (using storywright)',
+  'test-perf': 'run performance tests',
+  e2e: 'run e2e/component tests for the project',
+  'verify-packaging': 'verify packaging assets that will be shipped to npm',
+  // non default targets - not defined in nx.json
+  format: 'format code with prettier',
+  'code-style': 'format code with prettier and run eslint',
+  clean: 'remove build artifacts',
+  start: 'start the project',
+  // special target to trigger `nx show project <project>` cmd
+  help: 'show project help',
+};
+const defaultOmitTargets = ['nx-release-publish'];
+
+main().catch(e => {
+  console.error(e);
+  process.exit(1);
+});
+
+async function main() {
+  await cli({
+    projectTitle: projectTitleDefault,
+    projectHeader: projectHeaderDefault,
+    omitTargets: [...defaultOmitTargets, 'just'],
+    projectDescriptionFactory: createProjectDescription,
+    targetDescriptionFactory: createTargetDescription,
+  });
+}
+
+// ========================
+type ProjectHeader = { projectNameCell: string; descriptionCell: string };
+type ProjectDescriptionFn = (projectConfig: ProjectGraphProjectNode) => string;
+type TargetDescriptionFn = (
+  projectConfig: ProjectGraphProjectNode,
+  targetName: string,
+  targetsDescription: Record<string, string>,
+) => string;
+
+export async function cli(options: {
```

Comment:
this initial implementation is already generic - will be leveraged as a follow-up https://github.com/microsoft/fluentui/issues/32192